pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeTableDataManager.java

```diff
@@ -212,6 +216,63 @@ protected void doShutdown() {
     if (_leaseExtender != null) {
       _leaseExtender.shutDown();
     }
+    // Make sure we do metric cleanup when we shutdown the table.
+    if (_maximumPinotConsumptionDelayTracker != null) {
+      _maximumPinotConsumptionDelayTracker.shutdown();
+    }
+    // Now that segments can't report metric destroy metric for this table
+    _serverMetrics.removeTableGauge(_tableNameWithType, ServerGauge.MAX_PINOT_CONSUMPTION_DELAY_MS);
+  }
+
+  /*
+   * Pass-through method to handle CONSUMING->DROPPED transition.
+   *
+   * @param pid Partition id that we must stop tracking on this server.
+   */
+  public void stopTrackingPartitionDelay(long pid) {
+    if (_maximumPinotConsumptionDelayTracker != null) {
+      _maximumPinotConsumptionDelayTracker.stopTrackingPinotConsumptionDelayForPartition(pid);
+    }
+  }
+
+  /*
+   * Pass-through method to handle CONSUMING->ONLINE transition.
+   * If no new consumption is noticed for this segment in some timeout, we will read
+   * ideal state to verify the partition is still hosted in this server.
+   *
+   * @param pid partition id of partition to be verified as hosted by this server.
+   */
+  public void markPartitionForVerification(long pid) {
+    if (_maximumPinotConsumptionDelayTracker != null) {
+      _maximumPinotConsumptionDelayTracker.markPartitionForConfirmation(pid);
+    }
+  }
+
+  /*
+   * Pass-through function used by LLRealtimeSegmentManagers to update their partition delays
+   *
+   * @param d Ingestion delay being reported.
+   * @param t Timestamp of the measure being provided, i.e. when this delay was computed.
+   * @param pid Partition ID for which delay is being updated.
+   */
+  public void updatePinotIngestionDelay(long d, long t, long pid) {
+    if (_maximumPinotConsumptionDelayTracker != null) {
+      _maximumPinotConsumptionDelayTracker.recordPinotConsumptionDelay(d, t, pid);
+    }
+  }
+
+  /**
+   * Reads ideal state and verifies if current partition is hosted by the current server.
+   */
+  public boolean isPartitionHostedInThisServer(long partitionId) {
```

Comment:
Let us be consistent in naming ... partitionGroupId?

Reply:
There are diverging names across the code but I have renamed to partitionGroupId as that seems dominant